### ISSUE TITLE
Problem with TestComponentBuilder.prototype.createAsync()

### ISSUE THREAD

**Issue by svi3c** [angular/angular#5731]
Hi,

I just set up a component test similar to [border-component_test.ts](https://github.com/juliemr/ng2-test-seed/blob/master/src/test/border-component_test.ts) from ng2-test-seed (uses angular2@2.0.0-alpha.44). It works fine with angular2@2.0.0-alpha.46, but after an update to angular2@2.0.0-alpha.47 or angular2@2.0.0-alpha.48 the inner closure with the assertions is not called any more.
Here is my spec:
```js
import {ProgressBarComponent} from "./progressBar"
import {Component} from "angular2/angular2"
import {it, describe, expect, injectAsync, TestComponentBuilder, beforeEachProviders} from "angular2/testing"

@Component({
  template: '',
  directives: [ProgressBarComponent]
})
class TestComponent {
}

describe("ProgressBarComponent", () => {

  it("should have a progress-indicator which has the correct scale", injectAsync([TestComponentBuilder], (tcb) => {
    console.log("foo");
    return tcb
      .overrideTemplate(TestComponent, `<progress-bar value="10"></progress-bar>`)
      .createAsync(TestComponent)
      .then((fixture) => {
        console.log("bar");

        var indicator = fixture.debugElement.nativeElement.querySelector(".progress-indicator");

        expect(indicator.style.transform).toEqual(`scaleX(${10 / 100})`);
        //expect(indicator).toHaveCssStyle(`transform: scaleX(${10 / 100})`)
    })
  }));

});
```

What also changed in the update is the location of the browser_adapter module:
```js
// import {BrowserDomAdapter} from "angular2/src/core/dom/browser_adapter"
// Location changed in angular2@2.0.0-alpha.47
import {BrowserDomAdapter} from "angular2/src/platform/browser/browser_adapter"

BrowserDomAdapter.makeCurrent();
```

Also `injectAsync()` was deprecated and I tried to use `inject()` instead, but "bar" still won't show up in the console.

Am I missing something?

I'm using karma-jspm to run the tests.

**Comment by robwormald** [comment]
does a simple sanity check test run? eg 
```
describe('some stuff' , () => {
  
  it('should work' , () => {
    expect(true).Equal(true)
  });
  
});

```

**Comment by svi3c** [comment]
@robwormald Thanks for the quick reply!

Yes, the sanity check tests run.

I just updated the spec:

```js
import {Component} from "angular2/angular2"
import {it, describe, expect, injectAsync, TestComponentBuilder} from "angular2/testing"

@Component({
  template: '',
})
class TestComponent {
}

describe("injectAsync", function () {

  it("should wait for the promise and fail with wrong assertion", injectAsync([TestComponentBuilder], (tcb) => {
    return tcb
      .createAsync(TestComponent)
      .then(() => {
        expect(true).toEqual(false);
      })
  }), 1000);

});

describe("some stuff" , () => {

  it("should work" , () => {
    expect(true).toEqual(true)
  });

  it("should fail" , () => {
    expect(true).toEqual(false)
  });

});
```

The output:
```
09 12 2015 13:03:46.635:INFO [karma]: Karma v0.13.15 server started at http://localhost:8877/
09 12 2015 13:03:46.643:INFO [launcher]: Starting browser Chrome
09 12 2015 13:03:47.381:INFO [Chrome 46.0.2490 (Linux 0.0.0)]: Connected on socket lnjTFrMPiQg9pxQwAAAA with id 77778121
..
Chrome 46.0.2490 (Linux 0.0.0) some stuff should fail FAILED
	Expected true to equal false.
	    at Object.<anonymous> (http://localhost:8877/scripts/modules/progress/progressBar.spec.ts!transpiled:43:44)
Chrome 46.0.2490 (Linux 0.0.0): Executed 3 of 3 (1 FAILED) (0.102 secs / 0.028 secs)
```

Only one of the two wrong assertions fails. The async assertion is missed.

**Comment by svi3c** [comment]
Okay, it seems not to be a problem with `inject()`.
The following test fails as expected:

```js
import {it, describe, expect, inject} from "angular2/testing"

describe("inject", function () {

  it("should wait for the promise and fail with wrong assertion", inject([], () => {
    return new Promise((resolve) => {
      expect(true).toEqual(false);
      resolve();
    });
  }));

});
```

Maybe there is a problem in `TestComponentBuilder.prototype.createAsync()` and the Stacktrace is not logged?
Do you have any suggestions on retrieving the stack trace of the async function? I'm quite new to Angular2, ES6-Promises and TypeScript.

**Comment by goodfield** [comment]
Think it's related to https://github.com/angular/angular/issues/5662 and https://github.com/angular/angular/issues/5698.

Since alpha 48 my tests also seem to have some kind of synchronisation error ... some tests don't get executed, others get wrong injections.

**Comment by cexbrayat** [comment]
For the info, `injectAsync` is now deprecated, you can simply use `inject` (and the `return` becomes useless).
You can sometime have more info with a `.catch(err => console.log(err))` in your test.
FYI, I migrated to alpha-48 a few hundred tests with no major issues.

**Comment by goodfield** [comment]
@cexbrayat try to solve this one: http://plnkr.co/edit/ZOeFTO?p=preview ;-)

**Comment by cexbrayat** [comment]
:) `templateUrl` is broken right now, but you can override the template in your test with `tcb.overrideTemplate('<h1>fake tempalte</h1>').createAsync(TestCmp)`

**Comment by svi3c** [comment]
I just copied the specs from the plunker. And the async test is not failing.
So I assume that it has something to do with my project setup.

**Comment by goodfield** [comment]
@svi3c wanted to show the error with my plunker, running it at plunker the async doesn't fail either. Can't really reproduce you error ... but could it be, that it is caused by another spec ? so does it also happens if you just run your example without any other specs?

**Comment by svi3c** [comment]
@goodfield Sorry, I didn't put this correctly: I added the `.replaceTemplate()` call and it ran as expected in the plunker. Then I copied it.
The spec was the only one in the project at the time I ran it.

I just pushed the current state of the project containing the setup to github.
The spec which tests this behaviour is placed at https://github.com/frissdiegurke/sorhub-lib-frontend/blob/master/tests/unit/componentCompiler.spec.ts#L17
If you install deps with `npm i && jspm i`, change `xit()` to `it()`, run `jspm i angular2=angular2@2.0.0-alpha.50` then the test should unexpectedly pass.
